### PR TITLE
Make it harder to spoof the detection

### PR DESCRIPTION
**Pull request by pygy**
Just in case they wanted to pull another `document.all`...

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
-if (!Array.prototype.smoosh) {
+try {[].smooch()} catch(_) {
```
Comment (ckknight) [inline index.js]: Typo: `smooch` => `smoosh`.

File: index.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
   Array.prototype.smoosh = function smoosh() {
     return this[0];
   };
 }
 
-if (!Array.prototype.smooshMap) {
+try {[].smoochMap(function(){return []})} catch(_) {
```
Comment (ckknight) [inline index.js]: Same issue here - `smoochMap` => `smooshMap`.

Also, why is`smooshMap` doing `return []`? The `smooshMap` spec is perfectly happy if `undefined` is returned, so there should be no need for a function body at all.

File: index.js (hunk 1 above)
Comment (jamonholmgren) [reply index.js]: Shoudl be aliased to `[].😘`.

**Comment by pygy** [comment]
Good catch @ckknight

**Comment by staltz** [comment]
Thanks !

**Pull request merged**